var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -207,6 +207,10 @@ def setup_dependent_package(self, module, dep_spec):
         self.spec.mpicxx = join_path(self.prefix.bin, 'mpicxx')
         self.spec.mpifc  = join_path(self.prefix.bin, 'mpif90')
         self.spec.mpif77 = join_path(self.prefix.bin, 'mpif77')
+        if '+slurm' in self.spec:
+            self.spec.run = join_path(self.prefix.bin, 'srun')
```

Comment:
@alalazo i suppose 
```
self.spec['mpi'].run=which('srun')
``` 
returns executable, which is not the same as the string 
```
self.spec['mpi'].run=join_path(self.prefix.bin, 'mpirun')
```
? 

I intend to use it as `Executable('%s' % self.spec['mpi'].run)`.

p.s. unfortunately, i can't really test this part as i only have an access to macOS.